tests/test_flows.py

```diff
@@ -201,6 +201,70 @@ def check_cmdu_has_tlvs_exact(
             result = False
         return result
 
+    def check_topology_notification(self, eth_src: str, neighbors: list,
+                                    sta: env.Station, event: env.StationEvent, bssid: str) -> bool:
+        """Verify topology notification reliable multicast - given a source mac and
+           a list of neighbors macs, check that exactly one relayed multicast CMDU
+           was sent to the IEEE1905.1 multicast MAC address, and a single unicast
+           CMDU with the relayed bit unset to each of the given neighbors destination MACs.
+           Verify correctness of the association event TLV inside the topology notification.
+           Mark failure if any of the above conditions isn't met.
+
+        Parameters
+        ----------
+
+        eth_src: str
+            source AL MAC (origin of the topology notification)
+
+        neighbors: list
+            destination AL MACs (destinations of the topology notification)
+
+        sta: environment.Station
+            station mac
+
+        event: environment.StationEvent
+            station event - CONNECTED / DISCONNECTED
+
+        bssid: str
+            bssid Multi-AP Agent BSSID
+
+        Returns:
+        bool
+            True for valid topology notification, False otherwise
+        """
+        multicasts = self.check_cmdu_type("topology notification", 0x1, eth_src)
```

Comment:
Since you checkpoint, there should be only a single topology notification, so you can use check_cmdu_type_single instead.

Reply:
Correct, but `check_topology_notification` can't rely that the caller will checkpoint nor should it - we can easily support the same without checkpointing, so why not do that?

Reply:
OK